style: apply /style-guide pass to models/tables (#2725)
wandb/docs: models/tables/tables-download.mdx

---
description: "Export W&B Table data to pandas DataFrames and CSV files for offline analysis and data processing."
title: Export table data
keywords: ["pandas DataFrame", "CSV export", "get_dataframe", "table to artifact"]
---

Like all W&B artifacts, you can convert Tables into pandas DataFrames for data export. This page shows you how to export the data in a W&B Table to a pandas DataFrame and then to a CSV file, so that you can analyze or process the data outside of W&B.

## Convert a table to an artifact

> First, convert the table to an artifact so that you can retrieve it as a `Table` object to manipulate. To do this, use `artifact.get(table, "table_name")`:

Author: Fixed in 7910ed9. The prose now matches the example code: use `artifact.add(table, "my_table")` to add the table to the artifact and `artifact.get("my_table")` to retrieve it. The old `artifact.get(table, "table_name")` was wrong on both counts (it should be `add` to store, and `get` takes just the name).

title: Export table data
---

> Like all W&B artifacts, you can convert Tables into pandas DataFrames for data export. This page shows you how to export the data in a W&B Table to a pandas DataFrame and then to a CSV file, so that you can analyze or process the data outside of W&B.

Reviewer: ```suggestion
This page shows you how to export the data in a W&B Table to a pandas DataFrame and then to a CSV file, so that you can analyze or process the data outside of W&B.
```

Author: Dropped the leading sentence per the suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## Convert a table to an artifact

> First, add the table to an artifact and log it, then retrieve the table as a `Table` object you can manipulate. Use `artifact.add(table, "my_table")` to add the table to the artifact and `artifact.get("my_table")` to retrieve it later:

Reviewer: ```suggestion
1. Add the table to an artifact and log it. <code block>
2. Retrieve the table as a `Table` object you can manipulate.  <codeblock>
3. Use `artifact.add(table, "my_table")` to add the table to the artifact. <codeblock>
4. Use `artifact.get("my_table")` to retrieve the table. <codeblock>
```
This would be more readable as a procedure, I think.

Author: Reformatted the section as a two-step procedure with a code block per step.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

title: Export table data
---

This page shows you how to export the data in a W&B Table to a pandas DataFrame and then to a CSV file, so that you can analyze or process the data outside of W&B.

## Convert a table to an artifact

> To work with a logged table as a `Table` object, add it to an artifact, log the artifact, then retrieve the table:

Reviewer 2: Note that the uppercase "Table" is correct here because it refers to an instance of Python Class.

```suggestion
To work with a logged table as a Table object, add the table to an artifact, log the artifact, then retrieve the table from the artifact:
```

1. Add the table to an artifact with `artifact.add(table, "my_table")`, then log the artifact:

   ```python
   # Create and log a new table.
   with wandb.init() as r:
       artifact = wandb.Artifact("my_dataset", type="dataset")
       table = wandb.Table(
           columns=["a", "b", "c"], data=[(i, i * 2, 2**i) for i in range(10)]
       )
       artifact.add(table, "my_table")
       wandb.log_artifact(artifact)
   ```

> 2. Retrieve the table as a `Table` object with `artifact.get("my_table")`:

Reviewer 2: ```suggestion
2. Retrieve the table with `wandb.Artifact.get("my_table")`:
```

   ```python
   # Retrieve the created table using the artifact you created.
   with wandb.init() as r:
       artifact = r.use_artifact("my_dataset:latest")
       table = artifact.get("my_table")
   ```

## Convert the artifact to a DataFrame

> With the table retrieved from the artifact, convert it into a pandas DataFrame so that you can use DataFrame operations:

Reviewer 2: We already mentioned why we want to export it into a pandas DataFrame

```suggestion
With the table retrieved from the artifact, convert it into a pandas DataFrame:
```

```python
# Following from the last code example:
df = table.get_dataframe()
```

## Export data

> With your data in a DataFrame, you can export it using any method that pandas supports. For example, the following exports the data to a CSV file:

Reviewer 2: ```suggestion
With your data in a pandas DataFrame, you can export it using any method that pandas supports. For example, the following exports the data to a CSV file:
```
